package/MDAnalysis/core/groups.py

```diff
@@ -115,8 +115,15 @@
 from ._get_readers import get_writer_for, get_converter_for
 
 
-def _unpickle(u, ix):
-    return u.atoms[ix]
+def _unpickle(u, ix, cls=None):
+    if cls:
+        group = cls(ix, u)
+    else:
+        group = u.atoms[ix]
+        warnings.warn("You may be trying to read an AtomGroup that was pickled "
+        "with a pre 2.4 MDAnalysis version. If so, rewrite the pickle file to "
+        "silence this warning. More info at issue #3947 and PR #3953.", DeprecationWarning)
+    return group
```

Comment:
so a different solution to this would be to keep AtomGroup pickling going through this function (and not break anything) and create a new function (`_unpickle2`) which has the new signature.  This can then get tidied in 3.0 (which we *can* break compatibility with).

then you won't have the untested branch in the code

Reply:
Done.